text/0165-instrumentation-api.md

```diff
@@ -0,0 +1,236 @@
+# Instrumentation API
+
+API providing support to library instrumentation authors.
+
+## Motivation
+
+The OpenTelemetry API provides a means for data collection, for example starting / stopping
+spans or recording metrics. However, this is only half the story when it comes to instrumenting and
+misses out on common patterns required during instrumentation of libraries. Without defining these
```

Comment:
Can you please bring some examples of these patterns?

Reply:
Added a couple but let me know if more or more detail would help